### PR TITLE
Web scraping in Python

### PR DESCRIPTION
**Pull request by adithbharadwaj**
Requesting the team to review my article on web scraping using Python. It covers the basics of web scraping, beautifulsoup, and a hands-on example on scraping google forms. The folder contains 5 images: hero.jpg, excel.png, google-forms-inspect.png, hero.jpg, google-forms-class.png,  google-forms.png

### CODE REVIEW

**Comment by hectorkambow** [comment]
Good afternoon @glennslaven - 👋 could you please give this article a tech review whenever you are free? Thank you 🚀

**Comment by hectorkambow** [comment]
@adithbharadwaj we've started a tech review on your article - wanted to ask if you could make sure that this PR only has the Web scraping in Python article - each article should have their own (separate) branch - 👍

File: articles/web-scraping-in-python/index.md [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+The store_excel() function takes the data as a parameter and creates an excel sheet that contains the data. The Workbook() function creates a workbook with the name "content.xlsx" and add_worksheet() adds a new worksheet to our workbook. We then go through all the items in our data (all the questions) and add them to our worksheet using the write() method. The row and column integers denote the current row and column in the sheet.
+
+#### Putting it all together
+
+```python
+if __name__ == '__main__':
+
+	url = input('enter the url: ')
+	data = get_data(url)
+	print(data)
```
Comment (glennslaven) [inline articles/web-scraping-in-python/index.md:133]: Possibly redundant output of the div

File: articles/web-scraping-in-python/index.md [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+	url_client.close()
+	# beautiful soup parser to parse the files in html.
+	parser = soup(url_content, "html.parser")
+	content = parser.findAll('div', {'class': 'freebirdFormviewerComponentsQuestionBaseTitle'})
+	# print(parser.name_of_tag) to print the first element of that particular tag.
+
+	return content
+```
+
+The get_data function takes the URL as a parameter and returns all the elements that belong to a particular class. The urlopen() function establishes a connection to the URL and the read() function reads the contents of that page. We create an object called parser using beautifulsoup. This function takes two parameters: the contents of the page and the type of parser. In this case, we shall use the HTML parser. the findall() method finds all the HTML elements based on a given condition. In this case, we are looking for all the "div" tags that have the class "freebirdFormviewerComponentsQuestionBaseTitle".
```
Comment (glennslaven) [inline articles/web-scraping-in-python/index.md:95]:
```suggestion
The `get_data` function takes the URL as a parameter and returns all the elements that belong to a particular class. The `urlopen()` function establishes a connection to the URL and the `read()` function reads the contents of that page. We create an object called `parser` using beautifulsoup. This function takes two parameters: the contents of the page and the type of parser. In this case, we shall use the HTML parser. the `findall()` method finds all the HTML elements based on a given condition. In this case, we are looking for all the "div" tags that have the class "freebirdFormviewerComponentsQuestionBaseTitle".
```

Best to wrap any code in code tags so it's clear to the reader that you're referencing functions/variables.

File: articles/web-scraping-in-python/index.md [hunk 3]
```diff
@@ -0,0 +1,143 @@
...
+
+		text = str(item.text)
+		worksheet.write(row, col, text)
+		row += 1
+	# Finally, close the Excel file
+	# via the close() method.
+	workbook.close()
+```
+
+The store_excel() function takes the data as a parameter and creates an excel sheet that contains the data. The Workbook() function creates a workbook with the name "content.xlsx" and add_worksheet() adds a new worksheet to our workbook. We then go through all the items in our data (all the questions) and add them to our worksheet using the write() method. The row and column integers denote the current row and column in the sheet.
```
Comment (glennslaven) [inline articles/web-scraping-in-python/index.md:124]:
```suggestion
The `store_excel()` function takes the data as a parameter and creates an excel sheet that contains the data. The `Workbook()` function creates a workbook with the name "content.xlsx" and `add_worksheet()` adds a new worksheet to our workbook. We then go through all the items in our data (all the questions) and add them to our worksheet using the `write()` method. The row and column integers denote the current row and column in the sheet.
```

File: articles/web-scraping-in-python/index.md [hunk 4]
```diff
@@ -0,0 +1,143 @@
...
+if __name__ == '__main__':
+
+	url = input('enter the url: ')
+	data = get_data(url)
+	print(data)
+	store_excel(data)
+
+```
+
+In the main function, we take the URL as input from the user. We pass this URL to the get_data function that returns the elements that belong to the questions class. We then call the store_excel function and store the questions.
```
Comment (glennslaven) [inline articles/web-scraping-in-python/index.md:138]:
```suggestion
In the main function, we take the URL as input from the user. We pass this URL to the `get_data` function that returns the elements that belong to the questions class. We then call the `store_excel` function and store the questions.
```

**Comment by adithbharadwaj** [comment]
> @adithbharadwaj we've started a tech review on your article - wanted to ask if you could make sure that this PR only has the Web scraping in Python article - each article should have their own (separate) branch - +1

Sure @hectorkambow . I found that there was a merge conflict when the other article wasn't there. So I added it. Will remove it now.

File: articles/web-scraping-in-python/index.md (hunk 1 above)
Comment (adithbharadwaj) [reply articles/web-scraping-in-python/index.md]: Hey @glennslaven , I have made the changes you requested. Do you mean the print statement is redundant here? I have removed it.

**Comment by ninjaginja** [comment]
@adithbharadwaj Thanks for your contributions with this article. Unfortunately, upon further evaluation from our selection committee, we've made the decision not to publish this content. The Section product provides services that actively defend against web scraping, and therefore, publishing how-to content for this type of activity goes against what we do here. You're welcome to submit/publish to other forums, but we can't publish it on our properties. Thanks for your understanding.

**Pull request closed without merging**